src/Workspaces/Core/Portable/Storage/SQLite/SQLitePersistentStorageService.cs

```diff
@@ -126,20 +125,21 @@ protected override string GetDatabaseFilePath(string workingFolderPath)
                 }
                 throw;
             }
-
-            storage = sqlStorage;
-            return true;
         }
 
-        private static IDisposable TryGetDatabaseOwnership(string databaseFilePath)
+        private static IDisposable? TryGetDatabaseOwnership(string databaseFilePath)
         {
-            return IOUtilities.PerformIO<IDisposable>(() =>
+            return IOUtilities.PerformIO<IDisposable?>(() =>
             {
                 // make sure directory exist first.
                 EnsureDirectory(databaseFilePath);
 
+                var directoryName = Path.GetDirectoryName(databaseFilePath);
+                if (directoryName == null)
+                    return null;
```

Comment:
📝 This can't happen with a fully-qualified path. Should we just assert?

Reply:
i'd be ok with that.

Reply:
done.